### PR TITLE
Fix #10 

### LINKED ISSUE #10: Course filtering doesn't work for full-year courses
@nohackjustnoobb 

Full year courses have `idnumber` like "_ABCD1234_2024". These courses should not be filtered out regardless of the latest semester.

### PR DESCRIPTION
**Pull request by nohackjustnoobb**
Fix #10

### CODE REVIEW

File: lib/src/common/services/moodle_managers.dart [hunk 1]
```diff
@@ -87,29 +87,42 @@ class MoodleCourseManager with ChangeNotifier {
...
 
       for (final course in sortedCourses) {
-        var groups =
-            RegExp(r'^(.*)_(\w{0,3})_(\d{4})$').firstMatch(course.idnumber);
+        int? year, sem;
+        String? yearString, semString;
+
+        // One semester course
+        final oneSemesterGroups =
+            RegExp(r'^(.*)_(\w{1,3})_(\d{4})$').firstMatch(course.idnumber);
```
Comment (thermitex) [inline lib/src/common/services/moodle_managers.dart]: Subclass can be "ALL", suggest to change to (\w{1,4})

File: lib/src/common/services/moodle_managers.dart [hunk 2]
```diff
@@ -87,29 +87,42 @@ class MoodleCourseManager with ChangeNotifier {
...
+        if (oneSemesterGroups != null) {
```
Comment (thermitex) [inline lib/src/common/services/moodle_managers.dart:99]: Another corner case: _ABCD_ALL_2024 (which is a full-year course)
Better check if semString is valid as well

**Review by thermitex — CHANGES_REQUESTED** [review]

File: lib/src/common/services/moodle_managers.dart (hunk 2 above)
Comment (nohackjustnoobb) [reply lib/src/common/services/moodle_managers.dart:99]: I think it is already handled by the "int.tryParse". Or should I check if the sem is between 1 to 3?

Comment (thermitex) [reply lib/src/common/services/moodle_managers.dart:99]: Ah I see. Makes sense

**Review by thermitex — APPROVED** [review]

**Pull request merged**